### ISSUE TITLE
/srv/conda/envs/notebook/bin/python: No module named pip

### ISSUE THREAD

**Issue by jtpio** [jupyterhub/repo2docker#1131]
### Bug description

Opening a new issue as a follow-up to the comment posted in https://github.com/jupyterhub/repo2docker/pull/1062#issuecomment-1023073794.

Using the latest `repo2docker` (with `python -m pip install https://github.com/jupyterhub/repo2docker/archive/main.zip`), (existing) repos that have a custom `environment.yml` don't seem to be able to invoke `pip`, for example with `python -m pip`.

#### Expected behaviour

Running arbitrary `python -m pip install .` or similar should still be supported in a `postBuild` file.

#### Actual behaviour

Getting the following error:

```
/srv/conda/envs/notebook/bin/python: No module named pip
```

### How to reproduce

**With Binder**

Using the test gist: https://gist.github.com/jtpio/6268417579aaf252e06c72cd3ec29ebb

With `postBuild`:

```
python -m pip --help
```

And `environment.yml`:

```yaml
name: test
channels:
- conda-forge
dependencies:
- python >=3.10,<3.11
```

![image](https://user-images.githubusercontent.com/591645/151583923-c2110d69-473a-47e3-b767-fc75b5e5e361.png)

**Locally with repo2docker**

```
mamba create -n tmp -c conda-forge python=3.10 -y
conda activate tmp
python -m pip install https://github.com/jupyterhub/repo2docker/archive/main.zip
jupyter-repo2docker https://gist.github.com/jtpio/6268417579aaf252e06c72cd3ec29ebb
```

![image](https://user-images.githubusercontent.com/591645/151585579-60e888df-101d-4477-97f3-ef4b97c3e9c3.png)

### Your personal set up

Using this gist on mybinder.org: https://gist.github.com/jtpio/6268417579aaf252e06c72cd3ec29ebb

**Comment by jtpio** [comment]
Maybe related, the logs show a lot of `micromamba: command not found` errors:

![image](https://user-images.githubusercontent.com/591645/151587230-63ed5858-2bd6-43bc-b338-7ebd509b4b89.png)

**Comment by jtpio** [comment]
And a few:

```
The following arguments were not expected: /srv/conda/envs/notebook activate
Run with --help for more information.
```

![image](https://user-images.githubusercontent.com/591645/151587441-2c3f4bd5-f658-4629-9420-56c0da952b1d.png)

**Comment by minrk** [comment]
I _think_ this will be fixed by #1128 . Testing now.

**Comment by minrk** [comment]
Yes indeed, fixed by #1128

**Comment by jtpio** [comment]
Thanks!
